Winleafs.Wpf/Views/Layout/CreateEffectWindow.xaml.cs

```diff
@@ -0,0 +1,262 @@
+			ColorPicker.StandardColors.Clear();
+			_rgbToBrushMap = new Dictionary<uint, SolidColorBrush>();
+			
+			//Create a map of color to brush so that brushes can be reused
+			foreach (var rgb in colorsUsed)
+			{
+				AddToRgbToBrushMap(ColorFormatConverter.ToMediaColor(rgb));
+			}
+		}
+
+		private void AddToRgbToBrushMap(Color color)
+		{
+			ColorPicker.StandardColors.Add(new Xceed.Wpf.Toolkit.ColorItem(color, string.Empty));
+			_rgbToBrushMap.Add(ColorFormatConverter.ToRgb(color), color == Colors.Black ? Brushes.LightSlateGray : new SolidColorBrush(color));
+		}
+
+        private void Cancel_Click(object sender, RoutedEventArgs e)
+        {
+			//TODO Check for losing changes
+            Close();
+        }
+
+        private void SaveProgress_Click(object sender, RoutedEventArgs e)
+        {
+            UserSettings.Settings.ActiveDevice.CustomEffect = _customEffect;
+            UserSettings.Settings.SaveSettings();
+        }
+
+		private async void PlayOnDevice_Click(object sender, RoutedEventArgs e)
+		{
+			if (!float.TryParse(TransitionTextBox.Text, out var transitionSecs))
+			{
+				PopupCreator.Error(Layout.Resources.ValidTransistionTime);
+				return;
+			}
+
+			_customEffect.IsLoop = LoopCheckBox.IsChecked ?? false;
+			var customEffectCommandBuilder = new CustomEffectCommandBuilder(_customEffect);
+			var customEffectCommand = customEffectCommandBuilder.BuildDisplayCommand(transitionSecs);
+
+			var orchestrator = OrchestratorCollection.GetOrchestratorForDevice(UserSettings.Settings.ActiveDevice);
+
+			//TODO async so as not hang UI thread, but consider ContinueWith to handle errors
+			//which are lost by the use of async void
+			await orchestrator.ExecuteCustomEffectCommand(customEffectCommand);
+		}
+
+		private async void SaveToDevice_Click(object sender, RoutedEventArgs e)
+		{
+			//Always save progress as well
+			UserSettings.Settings.ActiveDevice.CustomEffect = _customEffect;
+			UserSettings.Settings.SaveSettings();
+
+			if (!float.TryParse(TransitionTextBox.Text, out var transitionSecs))
+			{
+				PopupCreator.Error(Layout.Resources.ValidTransistionTime);
+				return;
+			}
+
+			if (NameTextBox.Text.Trim().Length < 1)
+			{
+				PopupCreator.Error(Layout.Resources.ValidName);
+				return;
+			}
+			
+			_customEffect.IsLoop = LoopCheckBox.IsChecked ?? false;
+			var customEffectCommandBuilder = new CustomEffectCommandBuilder(_customEffect);
+			var customEffectCommand = customEffectCommandBuilder.BuildAddCommand(transitionSecs, NameTextBox.Text.Trim());
+
+			var orchestrator = OrchestratorCollection.GetOrchestratorForDevice(UserSettings.Settings.ActiveDevice);
+
+			//TODO async so as not hang UI thread, but consider ContinueWith to handle errors
+			//which are lost by the use of async void
+			await orchestrator.ExecuteCustomEffectCommand(customEffectCommand);
+		}
+
+		private void TransitionTextBox_PreviewTextInput(object sender, System.Windows.Input.TextCompositionEventArgs e)
```

Comment:
Use the using instead of specifying the full namespace

Reply:
Yeah, auto code creation did that for me and I was too lazy to change it! I'll add the `using`.